### PR TITLE
LPS-118626 use JSONUtil to create a string array instead of doing it by string concat

### PR DESCRIPTION
**Pull request by marcosapmf**
Related issue: https://issues.liferay.com/browse/LPS-118626

### CODE REVIEW

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by natocesarrego** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_8_1/UpgradeDDMFormFieldDataSourceType.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+						"dataSourceType", "[\"" + dataSourceType + "\"]");
+				}
+			}
+
+			JSONArray nestedFieldsJSONArray = jsonObject.getJSONArray(
+				"nestedFields");
+
+			if (nestedFieldsJSONArray != null) {
+				_upgradeFields(nestedFieldsJSONArray);
+			}
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_8_1/UpgradeDDMFormFieldDataSourceType.java:135]: Well done, @carolmariaabb! 😉

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_8_1/UpgradeDDMFormFieldDataSourceType.java [hunk 2]
```diff
@@ -119,7 +120,10 @@ private void _upgradeFields(JSONArray fieldsJSONArray) {
 					!dataSourceTypeString.endsWith(StringPool.CLOSE_BRACKET)) {
 
 					jsonObject.put(
-						"dataSourceType", "[\"" + dataSourceType + "\"]");
+						"dataSourceType",
+						JSONUtil.put(
+							dataSourceType
+						).toString());
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_8_1/UpgradeDDMFormFieldDataSourceType.java:126]: Good job, @marcosapmf!

**Comment by natocesarrego** [comment]
ci:forward

**Pull request closed without merging**